x/axelarnet/message_handler.go

```diff
@@ -287,7 +287,9 @@ func handleMessage(ctx sdk.Context, n types.Nexus, b types.BankKeeper, sourceAdd
 func handleMessageWithToken(ctx sdk.Context, n types.Nexus, b types.BankKeeper, sourceAddress nexus.CrossChainAddress, msg Message, token keeper.Coin) error {
 	id, txID, nonce := n.GenerateMessageID(ctx)
 
-	token, err := deductFee(ctx, b, msg.Fee, token, id)
+	destChain := funcs.MustOk(n.GetChain(ctx, nexus.ChainName(msg.DestinationChain)))
```

Comment:
reorder might be consensus breaking since user can trigger early failure? Shall we just use the lower cased `msg.destinationChain` instead for the event?

Reply:
good point